src/scripts/generate-binary-ops.py

```diff
@@ -0,0 +1,312 @@
+            if (random.randint(0, 1)):
+                op2 = random.randint(1, _constants.max)
+            else:
+                op2 = random.randint(_constants.min, - 1)
+        sum = op1 / op2
+        sum = int(sum) # Rounding because it might not be an integer
+        return {'oper': '/', 'op2': op2, 'sum': sum}
+
+    # Modulus
+    # Difference between % operator in python and js
+    # in python: -xxx % yy => yy - (abs(xxx) % yy)
+    # in js: -xxx % yy => - abs(xxx) % yy
+    def mod(self, op1, op2):
+        if op2 == 0:
+            if (random.randint(0, 1)):
+                op2 = random.randint(1, _constants.max)
+            else:
+                op2 = random.randint(_constants.min, -1)
+        if ((op1 * op2 < 0) and (op1 % op2 != 0)):
+            sum = ((op1 % op2) - op2)
+        else:
+            sum = (op1 % op2)
+        return {'oper': '%', 'op2': op2, 'sum': sum}
+
+    # Bitwise operations
+
+    # Bitshifting to left
+    def bitleftshift(self, op1, op2):
+        op2 = random.randint(_constants.bitmin, _constants.bitmax)
+        if op1 >= _constants.bitmax or op1 <= _constants.bitmin:
+            op2 = 0
+        sum = int(hex((op1 << ((op2 % (_constants.bitmax_exposant + 1)) & 0x1f))), 16)
+        if sum > _constants.bitmax or sum < _constants.bitmin: # Operand 2 must depend on the length of operand 1
+            if op1 != 0:
+                op2 = (_constants.bitmax_exposant - 1) - int.bit_length(op1)
+                if op2 < 0: # Operand 2 also cant go below zero
+                    op2 = 0
+        sum = int(hex((op1 << ((op2 % (_constants.bitmax_exposant + 1)) & 0x1f))), 16)
+        return {'oper': '<<', 'op2': op2, 'sum': sum}
+
+    # Bitshifting to right
+    # Python only can handle shiftcount between 0-31, but smaller and bigger numbers
+    # are also can be simulated with this method
+    def bitrightshift(self, op1, op2):
+        op2 = random.randint(_constants.bitmin, _constants.bitmax)
+        sum = int(hex((op1 >> ((op2 % (_constants.bitmax_exposant + 1)) & 0x1f))), 16)
+        return {'oper': '>>', 'op2': op2, 'sum': sum}
+
+    # Unsigned right bitshifting
+    def bitunsignedrightshift(self, op1, op2):
+        op2 = random.randint(_constants.uint32min, _constants.uint32max)
+        lnum = unsigned32(op1)
+        rnum = unsigned32(op2)
+        shiftcount = rnum & 0x1f
+        sum = ((lnum >> shiftcount) % _constants.uint32max)
+        return {'oper': '>>>', 'op2': op2, 'sum': sum}
+
+    # Bitwise and
+    def bitand(self, op1, op2):
+        sum = op1 & op2
+        return {'oper': '&', 'op2': op2, 'sum': sum}
+
+    # Bitwise or
+    def bitor(self, op1, op2):
+        sum = op1 | op2
+        return {'oper': '|', 'op2': op2, 'sum': sum}
+
+    # Bitwise xor (exclusive or)
+    def bitxor(self, op1, op2):
+        sum = op2 ^ op1
+        return {'oper': '^', 'op2': op2, 'sum': sum}
+
+# Generator function
+def generate_binary_tests(options):
```

Comment:
I was wondering why you do not put this function inside the class? And refactor it to smaller parts? I think you should give it a try, probably the readability will be improved with this.